src/frontend/src/pages/AdminToolsPage/BOMConfig/ManufacturerTable.tsx

```diff
@@ -24,12 +34,36 @@ const ManufacturerTable: React.FC = () => {
     return <ErrorPage message={manufacturersError?.message} />;
   }
 
-  const manufacturersTableRows = manufactureres.map((manufacturer) => (
+  const handleDeleteManufacturer = async (manufacturerName: string) => {
+    try {
+      deleteManufacturer(manufacturerName);
+      const updatedManufacturersTableRows = manufacturerList.filter(
+        (manufacturer) => manufacturer.name !== manufacturerName
+      );
+
+      setManufacturerList(updatedManufacturersTableRows);
+    } catch (error) {
+      console.error('Error deleting manufacturer:', error);
+    }
+  };
+
+  const manufacturersTableRows = manufacturerList.map((manufacturer) => (
     <TableRow>
       <TableCell align="left" sx={{ border: '2px solid black' }}>
         {datePipe(manufacturer.dateCreated)}
       </TableCell>
-      <TableCell sx={{ border: '2px solid black' }}>{manufacturer.name}</TableCell>
+      <TableCell sx={{ border: '2px solid black' }}>
+        <Grid container justifyContent="space-between">
+          <Grid sx={{ align: 'left' }}>{manufacturer.name}</Grid>
+          <Grid>
+            {manufacturerList.length > 1 ? (
```

Comment:
Ternary operation is not necessary here since we are mapping over each manufacturer in the list we want to create a row and a corresponding button for each manufacturer, dont need to check the size